clients/mock-client.rst

```diff
@@ -140,6 +140,88 @@ Or set a default exception::
         }
     }
 
+Mocking request-dependent responses
+-----------------------------------
+
+You can mock responses and exceptions which are only used in certain requests
+or act differently depending on the request.
+
+To conditionally return a response when a request is matched::
+
+    use Http\Mock\Client;
+
+    class YourTest extends \PHPUnit_Framework_TestCase
+    {
+        /**
+         * @expectedException \Exception
+         */
+        public function testClientThrowsException()
+        {
+            $client = new Client();
+
+            // $requestMatcher is an instance of Http\Message\RequestMatcher
+
+            $response = $this->createMock('Psr\Http\Message\ResponseInterface');
+            $client->on($requestMatcher, $response);
+
+            // $request is an instance of Psr\Http\Message\RequestInterface
+            $returnedResponse = $client->sendRequest($request);
+        }
+    }
+
+Or for an exception::
+
+    use Http\Mock\Client;
+
+    class YourTest extends \PHPUnit_Framework_TestCase
+    {
+        /**
+         * @expectedException \Exception
```

Comment:
i'd prefer dock to follow best practices and do `$this->expectException(\Exception::class);` in the code, on the line before sending the request.